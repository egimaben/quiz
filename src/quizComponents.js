import React from 'react';
import PropTypes from 'prop-types';
export const Header = () => {
    return (
      <header className="header"><h2>Author Quiz</h2>Do you know your authors?</header>
    );
  }
  
  export const Image = ({ src }) => {
    return <img className={'author-img animated-div'} src={src} alt='author of shown books' />
  }
  
  export const Book = ({ selectedBook, title, bookStyle, authorData, onRightAnswer, onWrongAnswer }) => {
    const style = selectedBook === title ? bookStyle : 'book';
    return <div className={style} onClick={() => {
      title === authorData.authoredBook
        ?
        onRightAnswer(title, authorData)
        :
        onWrongAnswer(title, authorData);
    }
    }>{title}</div>
  }
  
  export const Books = ({
    authorData,
    onWrongAnswer,
    onRightAnswer,
    bookStyle,
    selectedBook,
    showContinueButton,
    onContinue }) => {
  
    return <div className='books'>
      {
        authorData.bookOptions.map(
          (bookOption, index) => <Book
            onWrongAnswer={onWrongAnswer}
            onRightAnswer={onRightAnswer}
            bookStyle={bookStyle}
            selectedBook={selectedBook}
            key={index}
            title={bookOption} authorData={authorData} />
        )
  
      }
      {
        showContinueButton && <button onClick={onContinue} className='continue-button'>Continue</button>
      }
    </div>
  }
Books.propTypes={
  authorData:PropTypes.shape({
    bookOptions:PropTypes.arrayOf(PropTypes.string),
    authoredBook:PropTypes.string,
    avatar:PropTypes.string,
    authorName:PropTypes.string
  })
}
  
  export const Footer = () => {
    return <footer className='footer'>Copyright C</footer>
  }