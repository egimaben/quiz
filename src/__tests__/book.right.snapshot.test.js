'use strict'
import React from 'react';
import renderer from 'react-test-renderer';
import {Book} from '../quizComponents';
//when user selects the right book, the UI of the wrong book must have 'right-book' style
it('uses right selection/red highlight',()=>{
    const tree = renderer.create(
        <Book selectedBook='Animal farm' title='Animal farm' bookStyle='right-book'/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})