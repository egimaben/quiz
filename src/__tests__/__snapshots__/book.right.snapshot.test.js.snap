// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uses right selection/red highlight 1`] = `
<div
  className="right-book"
  onClick={[Function]}
>
  Animal farm
</div>
`;
